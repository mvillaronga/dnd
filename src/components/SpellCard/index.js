import React from 'react'

const SpellCard = ({name, level, onClick}) => {

    return <div onClick={onClick} >
        {name} - {level}
    </div>
}

export default SpellCard