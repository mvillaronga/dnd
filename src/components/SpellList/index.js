import React from 'react'


import SpellCard from '../SpellCard'

const SpellList = ({spells, onSpellClicked}) => {
        return (
        <div>
            {spells.map((spell, index) => (
                <SpellCard key={index} {...spell} onClick={() => onSpellClicked(spell)}/>
                ))}
        </div>
    )
}

export default SpellList