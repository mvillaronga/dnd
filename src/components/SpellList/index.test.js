import React from 'react';
import SpellList from '../SpellList';
import SpellCard from '../SpellCard'
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

configure({ adapter: new Adapter() });

test('Default Spell Card Rendering', () => {
  const component = renderer.create(
    <SpellList
      spells={[{ name: "Test Spell 1", level: "1" }, { name: "Test Spell 2", level: "9" }]}
      onSpellClicked={jest.fn()} />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('It should handle on click', () => {
  const click = jest.fn()
  const output = shallow(
    <SpellList
      spells={[{ name: "Test Spell 1", level: "1" }, { name: "Test Spell 2", level: "9" }]}
      onSpellClicked={click} />,
  )
  output.find(SpellCard).first().simulate('click')
  expect(click).toHaveBeenCalled()
})