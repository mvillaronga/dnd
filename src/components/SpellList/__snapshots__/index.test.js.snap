// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Spell Card Rendering 1`] = `
<div>
  <div
    onClick={[Function]}
  >
    Test Spell 1
     - 
    1
  </div>
  <div
    onClick={[Function]}
  >
    Test Spell 2
     - 
    9
  </div>
</div>
`;
